### PR TITLE
Room Details View

### PR DESCRIPTION
**Pull request by biancaflaidar**
- view, routing
- controller
- checking to see if user is logged in and if room exists

Tested with login:
something@something.com
password: something

### CODE REVIEW

File: app/scripts/app.js [hunk 1]
```diff
@@ -45,6 +45,10 @@ angular
         templateUrl: 'views/login.html',
         controller: 'LoginController'
       })
+      .when('/roomDetails',{
```
Comment (shakeelmohamed) [inline app/scripts/app.js]: Can we change this URL to be `/myRooms/:roomID`? Then you should be able to remove the hardcoded room ID and go directly to `/myRooms/-JoQjZMS9oMuTV5EdJXr` or whatever

File: app/views/roomDetails.html [hunk 2]
```diff
@@ -0,0 +1,40 @@
+<div class="container" ng-controller="DetailsController">
+    <ul class="list-group">
+        <li class="list-group-item">
+            <div ng-bind-html="myRoom.name"></div>
+        </li>
+        <li class="list-group-item">
+            <!-- TODO: implement room photo display here. -->
+            <img src="https://flavorwire.files.wordpress.com/2012/02/giraffe.jpeg" class="img-responsive" alt="Room Photo" style="width:304px;height:228px">
```
Comment (shakeelmohamed) [inline app/views/roomDetails.html]: Nice.

File: app/scripts/app.js (hunk 1 above)
Comment (biancaflaidar) [inline app/scripts/app.js]: Done!

File: app/scripts/controllers/roomDetails.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+                            description: roomSnapshot.val().description,
+                            maxGuests: roomSnapshot.val().maxGuests,
+                            name: roomSnapshot.val().name,
+                            nonSmoking: roomSnapshot.val().nonsmoking,
+                            petsAllowed: roomSnapshot.val().petsAllowed,
+                            roomPrice: roomSnapshot.val().roomPrice
+                        };
+                    });
+                    roomDNE = false;
+                    return true;
```
Comment (shakeelmohamed) [inline app/scripts/controllers/roomDetails.js]: Does this have to return `true`? If so, why? If not, change it to `return;`

File: app/scripts/controllers/roomDetails.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        var roomDNE = true;
+
+        var ref = new Firebase('https://test-admin-accounts.firebaseio.com/hotels/private_' + $rootScope.id + '/rooms');
+        ref.on("value", function(snapshot) {
+            snapshot.forEach(function(roomSnapshot) {
+               var key = roomSnapshot.key();
+               if (key == $routeParams.roomID)
+               {
+                    $scope.$apply(function() {
+                        $scope.myRoom = {
```
Comment (shakeelmohamed) [inline app/scripts/controllers/roomDetails.js]: Add a TODO here to making sure the properties exist. You can probably just do `$scope.myRoom = roomSnapshot.val();`

File: app/scripts/controllers/roomDetails.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+                    return true;
+               }
+            });
+            if (roomDNE)
+            {
+                console.log('Room does not exist.');
+                $location.path('/'); // TODO: change this to $location.path('/...'); (back to room list view)
+            }
+        }, function(errorObject) {
+            console.log("the read failed: "+ errorObject.code);
```
Comment (shakeelmohamed) [inline app/scripts/controllers/roomDetails.js]: Can we get a better error message here? And log the whole errorObject, not just the code

**Comment by shakeelmohamed** [comment]
Almost there, left a couple more comments

File: app/scripts/controllers/roomDetails.js (hunk 3 above)
Comment (biancaflaidar) [inline app/scripts/controllers/roomDetails.js]: According to Firebase documentation, you need to return true, it's special to snapshot.forEach().

File: app/scripts/controllers/roomDetails.js (hunk 4 above)
Comment (biancaflaidar) [inline app/scripts/controllers/roomDetails.js]: Already tried that.. the html doesn't get updated.

File: app/scripts/controllers/roomDetails.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+            $location.path('/');
+        }
+
+        var roomDNE = true;
+
+        var ref = new Firebase('https://test-admin-accounts.firebaseio.com/hotels/private_' + $rootScope.id + '/rooms');
+        ref.on("value", function(snapshot) {
+            snapshot.forEach(function(roomSnapshot) {
+               var key = roomSnapshot.key();
+               if (key == $routeParams.roomID)
```
Comment (shakeelmohamed) [inline app/scripts/controllers/roomDetails.js]: Should be `===`

File: app/scripts/controllers/roomDetails.js (hunk 4 above)
Comment (shakeelmohamed) [inline app/scripts/controllers/roomDetails.js]: That seems like a bug, put a comment for now and we'll go back to it

Comment (biancaflaidar) [inline app/scripts/controllers/roomDetails.js]: Oh it actually worked this time! I probably didn't do this inside $apply last time I tested. Thanks for catching that!

**Comment by biancaflaidar** [comment]
Updated

**Comment by shakeelmohamed** [comment]
Great, thanks!

**Pull request merged**